src/rgw/rgw_tracer.cc

```diff
@@ -0,0 +1,117 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "rgw_tracer.h"
+#include <yaml-cpp/yaml.h>
+
+namespace tracing {
+
+  const std::shared_ptr<opentracing::Tracer> Tracer::noop_tracer = opentracing::MakeNoopTracer();
+  TracerConfiguration tracing::Tracer::tracer_conf;
+
+  Tracer::Tracer(jaegertracing::Config& conf):open_tracer(jaegertracing::Tracer::make(conf)) {};
+
+  Tracer::Tracer(const std::string& config_file_path) {
+    auto config_yaml = YAML::LoadFile(config_file_path);
+    auto config = jaegertracing::Config::parse(config_yaml);
+    open_tracer =  jaegertracing::Tracer::make(config);
+  };
+
+  Tracer::~Tracer() {open_tracer->Close();}
+
+  std::unique_ptr<Span> Tracer::start_trace(const std::string& trace_name) {
+    std::unique_ptr<opentracing::Span> span;
```

Comment:
instead of using a local variable, just return the span created by the tracer in line 25 and line 28